src/main/java/net/minecraftforge/common/ISpecialArmor.java

```diff
@@ -139,11 +156,11 @@ else if (stack.getItem() instanceof ItemArmor && !source.isUnblockable())
                     }
                     ratio += prop.AbsorbRatio;
 
-                    double absorb = damage * prop.AbsorbRatio;
+                    double absorb = damage * prop.AbsorbRatio + (totalArmor > 0 ? absorbedDamage * prop.Armor / totalArmor : 0);
                     if (absorb > 0)
                     {
-                        ItemStack stack = inventory[prop.Slot];
-                        int itemDamage = (int)(absorb / 25D < 1 ? 1 : absorb / 25D);
+                        ItemStack stack = inventory[prop.Slot.getIndex()];
+                        int itemDamage = (int)(absorb < 1 ? 1 : absorb);
                         if (stack.getItem() instanceof ISpecialArmor)
```

Comment:
Well, we have two options: one is to have the ISpecialArmor state if its damage is applied before or after the vanilla calculation, and the other would be to have ISpecialArmor have the ability to apply its reduction before AND after the vanilla one. Which might be a headache to the modder, imo.